### PR TITLE
Administrator.py

### PR DESCRIPTION
**Pull request by sandip-narwade**
While adding a new user it is throwing "TypeError: Unicode-objects must be encoded before hashing"

### CODE REVIEW

**Comment by sandip-narwade** [comment]
encoded before hashing

**Comment by sandip-narwade** [comment]
encoded before hashing

**Comment by alessiosavi** [comment]
Hi @sandip-narwade, thank you for the pull request.
I've test your solution and i see no issue, only one questions before accept the pull request, cause that error is not raised running the script in my current environment: 
- Are you sure that you have installed `py-bcrypt==0.4` and not `bcrypt`?

File: datastructure/Administrator.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def get_name(self) -> str:
     def encrypt_password(plain_text_password: str) -> str:
         # Hash a password for the first time
         #   (Using bcrypt, the salt is saved into the hash itself)
-        return bcrypt.hashpw(plain_text_password, bcrypt.gensalt())
+        return bcrypt.hashpw(plain_text_password.encode('utf-8'), bcrypt.gensalt())
```
Comment (alessiosavi) [inline datastructure/Administrator.py]: Hi @sandip-narwade, thank you for the pull request.
I've test your solution and i see no issue, only one questions before accept the pull request, cause that error is not raised running the script in my current environment: 
- Are you sure that you have installed `py-bcrypt==0.4` and not `bcrypt`?

**Comment by alessiosavi** [comment]
I've create some unit test, and seems that the issue is not reproducible.

**Pull request closed without merging**